flair/datasets/entity_linking.py

```diff
@@ -1760,3 +2210,398 @@ def __init__(
+        url = "https://huggingface.co/datasets/bigbio/bc5cdr/resolve/main/CDR_Data.zip"
+
+        path = cached_path(url, data_folder)
+        data_path = data_folder / "CDR_Data" / "CDR.Corpus.v010516"
+        if not data_path.exists():
+            unpack_file(path, data_folder)
+            assert data_folder.exists()
+
+        results_files: Dict[str, Union[Path, List[Path]]] = {
+            "train": data_path / "CDR_TrainingSet.BioC.xml",
+            "dev": data_path / "CDR_DevelopmentSet.BioC.xml",
+            "test": data_path / "CDR_TestSet.BioC.xml",
+        }
+        return results_files
+
+    def _get_bioc_entity(self, span, db_id_key="MESH"):
+        offsets = [(loc.offset, loc.offset + loc.length) for loc in span.locations]
+
+        text = span.text
+
+        if len(offsets) > 1:
+            i = 0
+            texts = []
+            for start, end in offsets:
+                chunk_len = end - start
+                texts.append(text[i : chunk_len + i])
+                i += chunk_len
+                while i < len(text) and text[i] == " ":
+                    i += 1
+        else:
+            texts = [text]
+        db_ids = span.infons[db_id_key] if db_id_key else "-1"
+
+        # some entities are not linked and
+        # some entities are linked to multiple normalized ids
+        db_ids_list = [] if db_ids == "-1" else db_ids.split("|")
+
+        normalized = [{"db_name": db_id_key, "db_id": db_id} for db_id in db_ids_list]
+
+        return {
+            "id": span.id,
+            "offsets": offsets,
+            "text": texts,
+            "type": span.infons["type"],
+            "normalized": normalized,
+        }
+
+    def _file_to_dicts(self, filepath: Path) -> Iterator[Dict[str, Any]]:
+        reader = biocxml.BioCXMLDocumentReader(str(filepath))
+
+        for i, xdoc in enumerate(reader):
+            data = {
+                "document_id": xdoc.id,
+                "entities": [],
+                "passages": [],
+            }
+
+            char_start = 0
+            # passages must not overlap and spans must cover the entire document
+            for passage in xdoc.passages:
+                offsets = [[char_start, char_start + len(passage.text)]]
+                char_start = char_start + len(passage.text) + 1
+                data["passages"].append(
+                    {
+                        "type": passage.infons["type"],
+                        "text": [passage.text],
+                        "offsets": offsets,
+                    }
+                )
+
+            # entities
+            for passage in xdoc.passages:
+                for span in passage.annotations:
+                    ent = self._get_bioc_entity(span, db_id_key="MESH")
+                    if ent["type"].lower() == "chemical":
+                        data["entities"].append(ent)
+
+            yield data
+
+
+class BIGBIO_GNORMPLUS(BigbioCorpus):
```

Comment:
Also for each of the BigbioCorpus subclasses please add a brief description of the dataset and link.